File: lectures/code/cc_04_03.php
<?php

$a = array();

print("Testing php arrays\n");
$a["z"] = 8;
$a["z"] = 1;
$a["y"] = 9;
$a["b"] = 3;
$a["a"] = 4;
print_r($a);

printf("z=%d\n", $a["z"] ?? 42);
printf("x=%d\n", $a["x"] ?? 42);

printf("\nIterate\n");
foreach( $a as $k => $v ) {
    printf(" %s=%d\n", $k, $v);
}

// php cc_04_03.php

File: lectures/code/cc_04_03.java
import java.util.Map;
import java.util.TreeMap;

public class cc_04_03 {

    public static void main(String[] args)
    {   
        Map<String, Integer> map = new TreeMap<String, Integer> ();

        System.out.printf("Testing Map class\n");
        map.put("z", 8);
        map.put("z", 1);
        map.put("y", 2);
        map.put("b", 3);
        map.put("a", 4);
        System.out.println(map);

        System.out.println("z="+map.getOrDefault("z", 42));
        System.out.println("x="+map.getOrDefault("x", 42));

        System.out.printf("\nIterate\n");
        for (Map.Entry<String,Integer> entry : map.entrySet()) {
                System.out.println("Key = " + entry.getKey() + ", Value = " + entry.getValue());
        }

    }
}

// javac cc_04_03.java ; java cc_04_03 ; rm *.class

